ui/packages/catalog-ui-search/src/main/webapp/component/query-add/query-add.view.js

```diff
@@ -140,17 +135,12 @@ module.exports = Marionette.LayoutView.extend({
       })
     )
   },
-  showText() {
-    this.queryContent.show(
-      new QueryAdhoc({
-        model: this.model,
-      })
-    )
-  },
-  showBasic() {
+  showQueryForm(form) {
+    const options = form.options || {}
     this.queryContent.show(
-      new QueryBasic({
+      new form.view({
```

Comment:
Could we consume React components in our API instead?

Reply:
Because I included text, basic, and advanced search forms in the query form extension it will need to support both marionette and react. I've made it so react components can be passed in with `reactView` instead of `view`. not sure if there is a better way to differentiate between react/marionette